students/MicahBraun/Lesson 04/mailroom_2.py

```diff
@@ -0,0 +1,287 @@
+    print("Leaving menu...\n")
+    while True:
+        print("Enter the name of the person you are writing to (or enter 'list' to see a list of names or Q to quit) ")
+        fname_prompt = input("First Name: ").strip().capitalize()  # first name variable (strip any spaces)
+        if fname_prompt.upper() == "Q":  # if Q, then quit to menu()
+            menu()
+        elif fname_prompt.lower() == "list":
+            displaylist()
+        else:
+            lname_prompt = input("Last Name: ").strip().capitalize()  # last name variable
+            if lname_prompt.upper() == "Q":  # if Q, back to menu
+                menu()
+            elif lname_prompt.lower() == "list":  # if list, then list display
+                displaylist()
+            else:
+                key = fname_prompt + " " + lname_prompt
+                if key in donor_dict.keys():
+                    existing_donor = input(
+                        "That value is already in the list! Do you want to proceed with that selection? (Y/N): ")
+                    if existing_donor.upper() == "Y":  # if user proceeds, print display
+                        donation_amt = int(input("Enter in the donation amount from Donor {0}: $".format(key)))
+                        if str(donation_amt).upper() == "Q":
+                            menu()
+                        else:
+                            print(
+                                '{0} has donated ${1:,.2f}'.format(key, donation_amt))  # display name and donation amt
+                            int(donation_amt)
+                            get_index = 0  # index variable
+                            for item in range(0, len(donor_dict)):  # set index var to current name
+                                if list(donor_dict.values())[item][0] == key:
+                                    get_index = item
+                                    break
+                            firstname = list(donor_dict.values())[get_index][0].split(' ', ).pop(
+                                0)  # separate first name, create variable
+                            current_donations = int(
+                                list(donor_dict.values())[get_index][1])  # current amt/ var from dict{[val]}
+                            sum_donations = current_donations + donation_amt  # sum of all donations (current + new)
+                            list(donor_dict.values())[get_index][1] = float(sum_donations)  # update sum
+                            num_donations = int(list(donor_dict.values())[get_index][2]) + 1  # num donations = self + 1
+                            list(donor_dict.values())[get_index][2] = num_donations  # update num of donations
+                            list(donor_dict.values())[get_index][3] = averagedonations(sum_donations,
+                                                                                       num_donations)  # update avg
```

Comment:
You can use the key to directly access the values in the dict with the donor_dict[key] syntax, so you don't need to store the name as part of the list and access via `list(donor_dict.values())[get_index]`